### PR TITLE
Reload - Fix display ammo count in vehicle

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- title;
- ~~fix no `Check ammo` interaction on statics when CSW ammo handling is disabled;~~
- fix and improve `Take` EH;
- optimize fnc_displayAmmo.

When drag magazine to your weapon in gear in vehicle it will display empty ammo count. To fix it `weaponState` should be used instead of `currentWeapon`, `currentMuzzle` and `currentMagazine`.

### CODE REVIEW

File: addons/csw/functions/fnc_staticWeaponInit.sqf [hunk 1]
```diff
@@ -63,12 +63,13 @@ if (hasInterface && {!(_typeOf in GVAR(initializedStaticTypes))}) then {
         [_typeOf, 0, ["ACE_MainActions"], _disassembleAction] call EFUNC(interact_menu,addActionToClass);
     };
 
+    if (GVAR(ammoHandling) == 0) exitWith {};
```
Comment (PabstMirror) [inline addons/csw/functions/fnc_staticWeaponInit.sqf:66]: this is not the same

Comment (Dystopian) [reply addons/csw/functions/fnc_staticWeaponInit.sqf]: Yeah, my bad. Misread `||` instead of `&&`.

**Comment by Max255PL** [comment]
I'm guessing these shouldn't be happening? Current ACE master + CBA 3.15.1 only.

```16:55:09 Error in expression <_unit call ace_reload_fnc_displayAmmo}; \;>
16:55:09   Error position: <\;>
16:55:09   Error Invalid number in expression
16:55:09 Error in expression <_unit call ace_reload_fnc_displayAmmo}; \;>
16:55:09   Error position: <\;>
16:55:09   Error Invalid number in expression
16:55:09 Error in expression <_unit call ace_reload_fnc_displayAmmo}; \;},{}],true,[]],["CAManBase","Take",[{c>
16:55:09   Error position: <\;},{}],true,[]],["CAManBase","Take",[{c>
16:55:09   Error Invalid number in expression
16:55:09 Error in expression <_unit call ace_reload_fnc_displayAmmo}; \;},{}],true,[]],["CAManBase","Take",[{c>
16:55:09   Error position: <\;},{}],true,[]],["CAManBase","Take",[{c>
16:55:09   Error Invalid number in expression```

**Comment by Dystopian** [comment]
looks like that HEMTT bug with `\;` at the end. @Max255PL Did you use HEMTT?

**Comment by Max255PL** [comment]
Ah, indeed I did. I knew I saw something like this before but couldn't remember what was it about. Disregard then.

**Pull request merged**